pkg/models/monitoring/monitoring.go

```diff
@@ -89,21 +91,49 @@ func (mo monitoringOperator) GetMetricOverTime(expr, namespace string, start, en
 			return monitoring.Metric{}, err
 		}
 	}
-	return mo.c.GetMetricOverTime(expr, start, end, step), nil
+	return mo.prometheus.GetMetricOverTime(expr, start, end, step), nil
 }
 
 func (mo monitoringOperator) GetNamedMetrics(metrics []string, time time.Time, opt monitoring.QueryOption) Metrics {
-	ress := mo.c.GetNamedMetrics(metrics, time, opt)
+	ress := mo.prometheus.GetNamedMetrics(metrics, time, opt)
+	mr := mo.metricsserver.GetNamedMetrics(metrics, time, opt)
```

Comment:
If we make metricsserver an optional deployment, have to check if `metricsserver == nil` before calling

Reply:
We detected metrics API and stored into metricsAPIAvailable to judge metricsserver support.
https://github.com/Ma-Dan/kubesphere/blob/master/pkg/simple/client/monitoring/metricsserver/metricsserver.go#L41

Reply:
Fixed